### ISSUE TITLE
[Bug]: Ruby error on npm run dev after latest update

### ISSUE THREAD

**Issue by federicobenedetti** [Shopify/cli#1308]
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

Extension

### Expected behavior

Npm run dev doesn't exit after being run.

### Actual behavior

After issuing npm run dev, it will do everything just fine but it will eventually get stuck on bundle install (after instantiating ngrok tunnel and starting vite dev server).

### Stack trace

```shell
2023-02-07 17:21:45 | backend    |
2023-02-07 17:21:45 | backend    | > dev
2023-02-07 17:21:45 | backend    | > cross-env NODE_ENV=development nodemon index.js --ignore ./frontend
2023-02-07 17:21:45 | backend    |
2023-02-07 17:21:45 | frontend   |
2023-02-07 17:21:45 | frontend   | > dev
2023-02-07 17:21:45 | frontend   | > vite
2023-02-07 17:21:45 | frontend   |
2023-02-07 17:21:46 | backend    | [nodemon] 2.0.20
2023-02-07 17:21:46 | backend    | [nodemon] to restart at any time, enter `rs`
2023-02-07 17:21:46 | backend    | [nodemon] watching path(s): *.*
2023-02-07 17:21:46 | backend    | [nodemon] watching extensions: js,mjs,json
2023-02-07 17:21:46 | backend    | [nodemon] starting `node index.js`
2023-02-07 17:21:46 | frontend   |
2023-02-07 17:21:46 | frontend   |   vite v2.9.15 dev server running at:
2023-02-07 17:21:46 | frontend   |
2023-02-07 17:21:46 | frontend   |   > Network:  http://172.18.110.44:50049/
2023-02-07 17:21:46 | frontend   |   > Local:    http://localhost:50049/
2023-02-07 17:21:46 | frontend   |
2023-02-07 17:21:46 | frontend   |   ready in 930ms.
2023-02-07 17:21:46 | frontend   |
2023-02-07 17:21:47 | backend    | [shopify-api/INFO] version 6.1.0, environment Node v18.13.0

── external error ─────────────────────────────────────────────────────────────────────────────────────────────────

Error coming from `bundle install`

Command failed with exit code 6: bundle install
Could not find compatible versions

Because the current Bundler version (2.4.5) does not satisfy bundler ~> 2.3.11
  and Gemfile depends on bundler ~> 2.3.11,
  version solving has failed.

Your bundle requires a different version of Bundler than the one you're running.
Install the necessary version with `gem install bundler:2.3.26` and rerun
bundler using `bundle _2.3.26_ install`
Fetching gem metadata from https://rubygems.org/.........
Resolving dependencies...
```

### Reproduction steps

1. install latest shopify cli and themes version
2. run npm run dev

### Operating System

Windows 11

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.40.2

### Node version (run `node -v` if you're not sure)

18.13

### What language and version are you using in your application?

Node

**Comment by gonzaloriestra** [comment]
Hi! Thanks for creating the issue.

We have just released a new version (3.42) that should fix this issue. Could you please upgrade and try again?

**Comment by federicobenedetti** [comment]
@gonzaloriestra hey! yeah, i'll try asap, i'll let you know if it's fixed or not.
Thank you for the work btw :)
